### PR TITLE
completed fetch-in-react-practice exercise

### PR DESCRIPTION
**Pull request by AnthonyUrbina**
https://user-images.githubusercontent.com/109625982/198716941-67670e9a-d7b4-4504-ae51-d41188cea118.mov

### CODE REVIEW

File: fetch-in-react-practice/client/components/app.jsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+      body: JSON.stringify(todo)
+    }
+    )
+      .then(response => {
+        return response.json();
+      })
+      .then(data => {
+        const todosCopy = this.state.todos;
+        todosCopy[toggledTodoIndex] = data;
+        this.setState({ todos: todosCopy });
```
Comment (thebearingedge) [inline fetch-in-react-practice/client/components/app.jsx:121]: To make a legit copy you can use `.slice()` or `...` (spread)

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#description

**Review by thebearingedge — CHANGES_REQUESTED** [review]

File: fetch-in-react-practice/client/components/app.jsx (hunk 1 above)
Comment (AnthonyUrbina) [reply fetch-in-react-practice/client/components/app.jsx:121]: on it

File: fetch-in-react-practice/client/components/app.jsx [hunk 2]
```diff
@@ -116,8 +116,8 @@ export default class App extends React.Component {
         return response.json();
       })
       .then(data => {
-        const todosCopy = this.state.todos;
-        todosCopy[toggledTodoIndex] = data;
+        const todosCopy = [...this.state.todos];
+        todosCopy[toggledTodoIndex].isCompleted = todo.isCompleted;
```
Comment (thebearingedge) [inline fetch-in-react-practice/client/components/app.jsx:120]: This is actually modifying the original object in the array. Instead, replace it like you were already doing

```js
todosCopy[toggledIndex] = data;
```

The idea is to always be swapping objects completely. If that object is in an array, you will swap the array too, just leave the unaffected stuff in the copy.

**Review by thebearingedge — CHANGES_REQUESTED** [review]

File: fetch-in-react-practice/client/components/app.jsx (hunk 2 above)
Comment (AnthonyUrbina) [reply fetch-in-react-practice/client/components/app.jsx]: okay i think i got it right this time.

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**